src/coreclr/jit/codegenarmarch.cpp

```diff
@@ -1728,6 +1728,122 @@ void CodeGen::genCodeForIndexAddr(GenTreeIndexAddr* node)
     genProduceReg(node);
 }
 
+//------------------------------------------------------------------------
+// genGetVolatileLdStIns: Determine the most efficient instruction to perform a
+//    volatile load or store and whether an explicit barrier is required or not.
+//
+// Arguments:
+//    currentIns   - the current instruction to perform load/store
+//    indir        - the indirection node representing the volatile load/store
+//    needsBarrier - OUT parameter. Set to true if an explicit memory barrier is required.
+//
+// Return Value:
+//    instruction to perform the volatile load/store with.
+//
+instruction CodeGen::genGetVolatileLdStIns(instruction currentIns, GenTreeIndir* indir, bool* needsBarrier)
```

Comment:
Couldn't this function also do the `genIsValidIntReg(targetReg)` check instead of the callers?